picard/file_dilaog.py

```diff
@@ -0,0 +1,274 @@
+        self.cb.addItems(self.formats_show)
+        self.cb.resize(200,30)
+        self.cb.move(wi-.183*wi,he-.17*he)
+        self.cb.currentIndexChanged.connect(self.format_modifier)
+
+        self.resize(wi,he)
+        sys.exit(self.exec_())
+
+    def format_modifier(self,i):
+
+        self.formats.clear()
+        self.formats_lower.clear()
+        if not self.cb.itemText(i) == "All Supported Formats":
+            self.formats = [self.cb.itemText(i)]
+        else:
+            self.formats=["MP3","FLAC","OGG","M4A","WMA","WAV","MP4","AAC","flv"]
+
+        for i in self.formats:self.formats_lower.append(i.lower())
+
+        self.show_files(False)
+
+    def default_setting(self):
+        indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            #for index in (indexes):
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                self.open_action(indexes)
+            else:
+                self.select_action(indexes)
+        else :
+            self.select_action(indexes)
+
+    def open_action(self,indexes):
+        if indexes == True:
+            indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                hola = (self.table.item(indexes[0].row(),0)).text()[6:]
+                self.path = os.path.join(self.path,hola)
+                self.show_files(False)
+
+    def select_action(self,indexes):
+        if indexes == True:
+            indexes = self.table.selectionModel().selectedRows()
+        self.send_things = []
+        for index in (indexes):
+            hola = (self.table.item(index.row(),0)).text()[6:]
+            if os.path.isdir(os.path.join(self.path,hola)):
+                self.recu_folder(self.send_things,os.path.join(self.path,hola))
+            else:
+                self.send_things.append(os.path.join(self.path,hola))
+        for  i in self.send_things:
+            print(i)
+        #print(self.send_things)
+        sys.exit()
+
+    def recu_folder(self,lis,path):
+       temp = os.listdir(path)
+       for i in temp:
+            if os.path.isfile(os.path.join(path,i)):
+                for j in self.formats_lower:
+                    if i.endswith('.'+j):
+                        lis.append(os.path.join(path,i))
+            else:
+                self.recu_folder(lis,os.path.join(path,i))
+
+    def moveBack_action(self):
+        self.path = os.path.dirname((os.path.abspath(self.path)))
+        self.show_files(False)
+
+    def moveNext_action(self):
+        indexes = self.table.selectionModel().selectedRows()
+        if len(indexes) == 1:
+            if os.path.isdir(os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])):
+                self.path = os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])
```

Comment:
Too much code redundancy, cache `os.path.join(self.path,self.table.item(indexes[0].row(),0).text()[6:])` (and explain it...)